sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -72,6 +75,69 @@ abstract class StringRegexExpression extends BinaryExpression
   }
 }
 
+sealed abstract class LikeBase extends StringRegexExpression {
```

Comment:
@MaxGekk can you highlight the major code difference between this new `LikeBase` and the previous `Like`?

Reply:
I removed `LikeBase` and implemented `ILike` as `RuntimeReplaceable` to `Like`.